### PR TITLE
[CI/Build] bump minimum cmake version

### PR DESCRIPTION
**Pull request by dtrifiro**
- `find_python_from_executable` uses `find_package(PYTHON ... Development.SABIModule)`, which is only available [since cmake>=3.26](https://cmake.org/cmake/help/latest/module/FindPython.html)
- `cmake` and `ninja` are build-time requirements, so they should not be  in `requirements-common.txt` or `requirements-openvino.txt`

### CODE REVIEW

**Comment by dtrifiro** [comment]
Some extra context for this: having an older `cmake` version would result in a cryptic error when building

```CMake Error at cmake/utils.cmake:10 (message):
  Unable to find python matching:
  /home/dtrifiro/work/vllm/.venv/bin/python.
Call Stack (most recent call first):
  CMakeLists.txt:43 (find_python_from_executable)
```

**Review by bnellnm — APPROVED** [review]
LGTM!

**Comment by dtrifiro** [comment]
CI failures seem unrelated:

```
test_traces
...
ValueError: OpenTelemetry packages must be installed before configuring 'otlp_traces_endpoint'
```

**Review by mgoin — APPROVED** [review]

**Comment by dtrifiro** [comment]
Anything holding this back?

**Comment by mgoin** [comment]
@dtrifiro looking at the failing check logs, it seems the Intel CPU test cannot find its cmake executable with this change. It would be great if you could investigate this

**Comment by dtrifiro** [comment]
@mgoin I opened https://github.com/vllm-project/vllm/pull/7298 to take care of the `Dockerfile.cpu` issues, this should go through once that's merged.

**Comment by dtrifiro** [comment]
Looking at the failure logs, it seems that they're unrelated:

```
entrypoints/openai/test_completion.py::test_single_completion[-zephyr-pa-8] INFO 08-09 08:48:57 logger.py:36] Received request cmpl-c29cba85cf3a4c7a930f4aa562d54614-0: prompt: 'Hello, my name is', params: SamplingParams(n=1, best_of=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.0, top_p=1.0, top_k=-1, min_p=0.0, seed=None, use_beam_search=False, length_penalty=1.0, early_stopping=False, stop=[], stop_token_ids=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=5, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None), prompt_token_ids: [1, 22557, 28725, 586, 1141, 349], lora_request: None, prompt_adapter_request: PromptAdapterRequest(prompt_adapter_name='zephyr-pa', prompt_adapter_id=1, prompt_adapter_local_path='/root/.cache/huggingface/hub/models--swapnilbp--llama_tweet_ptune/snapshots/ad7faa5e642dd3f4311c927b0c423402444cd7a8', prompt_adapter_num_virtual_tokens=8).
[2024-08-09T08:48:57Z] INFO 08-09 08:48:57 async_llm_engine.py:199] Added request cmpl-c29cba85cf3a4c7a930f4aa562d54614-0.
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61] Engine background task failed
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61] Traceback (most recent call last):
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 51, in _log_task_completion
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     return_value = task.result()
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 772, in run_engine_loop
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     result = task.result()
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 715, in engine_step
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     request_outputs = await self.engine.step_async(virtual_engine)
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 282, in step_async
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     output = await self.model_executor.execute_model_async(
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/gpu_executor.py", line 159, in execute_model_async
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     output = await make_async(self.driver_worker.execute_model
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     result = self.fn(*self.args, **self.kwargs)
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/worker_base.py", line 236, in execute_model
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     self.model_runner.prepare_model_input(
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1435, in prepare_model_input
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     model_input = self._prepare_model_input_tensors(
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1006, in _prepare_model_input_tensors
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     builder.add_seq_group(seq_group_metadata)
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 638, in add_seq_group
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     per_seq_group_fn(inter_data, seq_group_metadata)
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 590, in _compute_prompt_adapter_input
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61]     assert inter_data.n_seqs == 1
[2024-08-09T08:48:57Z] ERROR 08-09 08:48:57 async_llm_engine.py:61] AssertionError
[2024-08-09T08:48:57Z] Exception in callback _log_task_completion(error_callback=<bound method...7fec46856bc0>>)(<Task finishe...ertionError()>) at /usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py:41
[2024-08-09T08:48:57Z] handle: <Handle _log_task_completion(error_callback=<bound method...7fec46856bc0>>)(<Task finishe...ertionError()>) at /usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py:41>
[2024-08-09T08:48:57Z] Traceback (most recent call last):
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 51, in _log_task_completion
[2024-08-09T08:48:57Z]     return_value = task.result()
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 772, in run_engine_loop
[2024-08-09T08:48:57Z]     result = task.result()
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 715, in engine_step
[2024-08-09T08:48:57Z]     request_outputs = await self.engine.step_async(virtual_engine)
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 282, in step_async
[2024-08-09T08:48:57Z]     output = await self.model_executor.execute_model_async(
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/gpu_executor.py", line 159, in execute_model_async
[2024-08-09T08:48:57Z]     output = await make_async(self.driver_worker.execute_model
[2024-08-09T08:48:57Z]   File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
[2024-08-09T08:48:57Z]     result = self.fn(*self.args, **self.kwargs)
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/worker_base.py", line 236, in execute_model
[2024-08-09T08:48:57Z]     self.model_runner.prepare_model_input(
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1435, in prepare_model_input
[2024-08-09T08:48:57Z]     model_input = self._prepare_model_input_tensors(
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1006, in _prepare_model_input_tensors
[2024-08-09T08:48:57Z]     builder.add_seq_group(seq_group_metadata)
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 638, in add_seq_group
[2024-08-09T08:48:57Z]     per_seq_group_fn(inter_data, seq_group_metadata)
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 590, in _compute_prompt_adapter_input
[2024-08-09T08:48:57Z]     assert inter_data.n_seqs == 1
[2024-08-09T08:48:57Z] AssertionError
[2024-08-09T08:48:57Z]
[2024-08-09T08:48:57Z] The above exception was the direct cause of the following exception:
[2024-08-09T08:48:57Z]
[2024-08-09T08:48:57Z] Traceback (most recent call last):
[2024-08-09T08:48:57Z]   File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
[2024-08-09T08:48:57Z]     self._context.run(self._callback, *self._args)
[2024-08-09T08:48:57Z]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 63, in _log_task_completion
[2024-08-09T08:48:57Z]     raise AsyncEngineDeadError(
[2024-08-09T08:48:57Z] vllm.engine.async_llm_engine.AsyncEngineDeadError: Task finished unexpectedly. This should never happen! Please open an issue on Github. See stack trace above for the actual cause.
[2024-08-09T08:48:57Z] WARNING 08-09 08:48:57 server.py:103] Failed to abort request cmpl-c29cba85cf3a4c7a930f4aa562d54614-0
```

**Review by njhill — APPROVED** [review]
Thanks @dtrifiro

**Pull request merged**
